ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/QueryResponseHandler.java

```diff
@@ -35,10 +52,32 @@
 
   @Override
   protected void doHandleBodyBuffer(final Buffer buff) {
+    final Buffer strippedValidJson = toJsonMsg(buff, true);
+
+    if (strippedValidJson.toString().equals("")) {
+      return;
+    }
+
     if (!hasReadArguments) {
-      handleArgs(buff);
+      handleArgs(strippedValidJson);
     } else {
-      handleRow(buff);
+      final JsonNode rowNode;
+      try {
+        rowNode = JSON_MAPPER.readTree(strippedValidJson.getBytes());
```

Comment:
Also, another question.  Is there any reason we used Jackson vs the Vertx json it was previously using?  Just curious if the handling of BigDecimals was the reason or something like that.

Reply:
yeah, handling `BigDecimal` becomes much cleaner with jackson